### PR TITLE
[HOTFIX] Old stores cannot read with new table infered through sdk.

### PR DESCRIPTION
**Pull request by ravipesala**
Problem: Old stores column schema is written in the different case then fileformat cannot read data because of sdk infer schema always gives lower case schema.
Solution: Do case insensitivity check while comparing.
It also disables prefetch as it is redundant for fileformat read and not getting inputmetrics properly if we use thread

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/211/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/380/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.3/8450/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/216/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/385/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.3/8455/

**Comment by ajantha-bhat** [comment]
@ravipesala : will it be better if we convert schema to lowercase just after inferring ?

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java [hunk 1]
```diff
@@ -165,14 +165,15 @@ private static boolean isColumnMatches(boolean isTransactionalTable,
     // column ID but can have same column name
     if (tableColumn.getDataType().isComplexType() && !(tableColumn.getDataType().getId()
         == DataTypes.ARRAY_TYPE_ID)) {
-      if (tableColumn.getColumnId().equals(queryColumn.getColumnId())) {
+      if (tableColumn.getColumnId().equalsIgnoreCase(queryColumn.getColumnId())) {
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: I think there are many other place columnID check is there. example filter scenario. 
If handle this way need to check and handle all the places.

I suggest better to convert inferred schema to lower case at the place of inferring. That way no need to handle all these places

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/220/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/389/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/394/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.3/8464/

File: core/src/main/java/org/apache/carbondata/core/metadata/CarbonMetadata.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ public CarbonDimension getCarbonDimensionBasedOnColIdentifier(CarbonTable carbon
     List<CarbonDimension> listOfCarbonDims =
         carbonTable.getDimensionByTableName(carbonTable.getTableName());
     for (CarbonDimension dimension : listOfCarbonDims) {
-      if (dimension.getColumnId().equals(columnIdentifier)) {
+      if (dimension.getColumnId().equalsIgnoreCase(columnIdentifier)) {
```
Comment (xuchuanyin) [inline core/src/main/java/org/apache/carbondata/core/metadata/CarbonMetadata.java]: Will the comparison for columnId still need case insensitive?
If it is yes, I'd suggest you to leave comments for this (for old store before version XXX).

File: core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java [hunk 3]
```diff
@@ -476,7 +476,11 @@ private BlockExecutionInfo getBlockExecutionInfoForBlock(QueryModel queryModel,
     blockExecutionInfo
         .setTotalNumberDimensionToRead(
             segmentProperties.getDimensionOrdinalToChunkMapping().size());
-    blockExecutionInfo.setPrefetchBlocklet(!queryModel.isReadPageByPage());
+    if (queryModel.isReadPageByPage()) {
```
Comment (xuchuanyin) [inline core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java]: It seems this modification is not related with the description of this PR

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java [hunk 4]
```diff
@@ -165,14 +165,15 @@ private static boolean isColumnMatches(boolean isTransactionalTable,
...
         return true;
       } else {
         return isColumnMatchesStruct(tableColumn, queryColumn);
       }
     } else {
-      return (tableColumn.getColumnId().equals(queryColumn.getColumnId()) || (!isTransactionalTable
-          && tableColumn.getColName().equals(queryColumn.getColName())));
+      return (tableColumn.getColumnId().equalsIgnoreCase(queryColumn.getColumnId()) || (
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: I think we should have more strict comparison in case of restructure.
If column with same name is added with different name with different data type, will be problem.
Let's use data type also to compare.

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java (hunk 1 above)
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: Same comment give below applicable here also.

File: core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java (hunk 3 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java]: Updated description

File: core/src/main/java/org/apache/carbondata/core/metadata/CarbonMetadata.java (hunk 2 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/metadata/CarbonMetadata.java]: this is the case for non-transactional flow. It uses column name for columnid as well. It is the safest way to do the check and there would not be any side effects on it,

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java (hunk 4 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: There would not be restructure for non transactional table. And dso the case sensitivity check only for safety.

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java (hunk 1 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: same as above comment

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java (hunk 4 above)
Comment (KanakaKumar) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: OK

**Comment by KanakaKumar** [comment]
LGTM

**Comment by kumarvishal09** [comment]
LGTM
